common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilterImpl.java

```diff
@@ -220,6 +220,35 @@ public BloomFilter mergeInPlace(BloomFilter other) throws IncompatibleMergeExcep
     return this;
   }
 
+  @Override
+  public BloomFilter intersectInPlace(BloomFilter other) throws IncompatibleMergeException {
+    // Duplicates the logic of `isCompatible` here to provide better error message.
+    if (other == null) {
```

Comment:
I wonder if we can just factor out these checks into a separate private utility method, to avoid the duplication?

Reply:
Makes sense, I just did that.